contracts/AINFT721.sol

```diff
@@ -0,0 +1,304 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721Enumerable.sol";
+import "@openzeppelin/contracts/security/Pausable.sol";
+import "@openzeppelin/contracts/access/AccessControl.sol";
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721Burnable.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+
+import "@openzeppelin/contracts/utils/Strings.sol";
+import "@openzeppelin/contracts/interfaces/IERC721.sol";
+
+import "./interfaces/IERC4906.sol";
+import "./interfaces/IAINFT.sol";
+
+/**
+ *@dev AINFT721 contract
+ */
+contract AINFT721 is
+    ERC721Enumerable,
+    Pausable,
+    AccessControl,
+    ERC721Burnable,
+    IERC4906,
+    IAINFT
+{
+    using Counters for Counters.Counter;
+    using Strings for uint256;
+
+    struct MetadataContainer {
+        address updater;
+        string metadataURI;
+    }
+    bool public immutable IS_CLONED;
+    IERC721 public immutable ORIGIN_NFT;
+    address public PAYMENT_PLUGIN;
+    bytes32 public constant PAUSER_ROLE = keccak256("PAUSER_ROLE");
+    bytes32 public constant MINTER_ROLE = keccak256("MINTER_ROLE");
+
+    Counters.Counter private _tokenIdCounter;
+    string private baseURI;
+    mapping(bytes32 => MetadataContainer) private _metadataStorage; // keccak256(bytes(tokenId, <DELIMETER>, version)) => MetadataContainer
+    mapping(uint256 => uint256) private _tokenURICurrentVersion; // tokenId => tokenURIVersion
+
+    constructor(string memory name_, string memory symbol_, bool isCloned_, address originNFT_) ERC721(name_, symbol_) {
+        if (isCloned_) {
+            //FIXME: need to use tx.origin?
+            _grantRole(DEFAULT_ADMIN_ROLE, tx.origin);
```

Comment:
what's different between `if(isCloned_) { ... /* HERE */ ... } else { ... /* THERE */ ... }`.
what's expected `tx.origin` value whether isCloned_ on/off?

Reply:
oh.. it's weird. I don't know why I use if condition. I'll remove it thanks!